File: AplicatieVanzariMasini-Front/src/app/_models/announce.ts
import { PhotoForAnnounce } from './photoForAnnounce';
import { Car } from './car';
import { User } from './user';

export interface Announce {
    announceId: number;
    title: string;
    description: string;
    createdDate: Date;
    features: string;
    carId: number;
    mainPhotoUrl?: string;
    km?: string;
    price?: string;
    enginePower?: string;
    cylindricalCapacity?: string;
    state?: string;
    damaged?: string;
    particleFilter?: string;
    rightHandDrive?: string;
    userId?: number;
    userName?: string;
    userMainPhotoUrl?: string;
    userCreated?: string;
    brand?: string;
    model?: string;
    modelVersion?: string;
    body?: string;
    fuel?: string;
    location: string;
    country?: string;
    transmission?: string;
    manufacturingDate?: string;
    gearbox?: string;
    pollutionRule?: string;
    car?: Car;
    photosForAnnounce?: PhotoForAnnounce[];
}

File: AplicatieVanzariMasini-Back/AplicatieVanzariMasini-Back/Models/Announce.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AplicatieVanzariMasini_Back.Models
{
    public class Announce
    {
        public int AnnounceId { get; set; }
        public int AnnounceView { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public string Features { get; set; }
        public int CarId { get; set; }
        public virtual Car Car { get; set; }
        public int UserId { get; set; }
        public virtual User User { get; set; }
        public virtual ICollection<PhotoForAnnounce> PhotosForAnnounce { get; set; }
        public virtual ICollection<SaveAnnounce> SaveAnnounces { get; set; }



    }
}
